nipap/nipap/authlib.py

```diff
@@ -327,17 +335,45 @@ def authenticate(self):
 
         # auth succeeded
         self.authenticated_as = self.username
-        self._authenticated = True
         self.trusted = False
         self.readonly = False
 
         try:
-            res = self._ldap_conn.search_s(self._ldap_basedn, ldap.SCOPE_SUBTREE, 'uid=' + self.username, ['cn'])
+            res = self._ldap_conn.search_s(self._ldap_basedn, ldap.SCOPE_SUBTREE, self._ldap_search.format(ldap.dn.escape_dn_chars(self.username)), ['cn','memberOf'])
```

Comment:
Never mind then. From a cursory look at the python-ldap it seemed like it would raise the NO_SUCH_ATTRIBUTE exception when you request an attribute that doesn't exist, but since it doesn't, it doesn't matter. I also overlooked that the search result was used to get the name of the user, so it seemed as if the result wasn't used at all if groups weren't non-empty. 

Just disregard my previous comment - doing just one search is fine.

Adding a specific except clause for the KeyError, that raises an appropriate exception would probably be a good idea though (since it indicates misconfiguration).